src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -189,8 +192,43 @@ class BaseOptionsSelector extends Component {
         return defaultIndex;
     }
 
+    /**
+     * When pagination is enabled,
+     * maps sections to render only allowed count of them.
+     */
+    get sections() {
+        if (!_.isNumber(this.props.itemsPerPage)) {
+            return this.props.sections;
+        }
+
+        return _.map(this.props.sections, (section) => {
+            if (_.isEmpty(section.data)) {
+                return section;
+            }
+
+            return {
+                ...section,
+                data: section.data.slice(0, this.props.itemsPerPage * this.state.paginationPage),
+            };
+        });
+    }
+
+    /**
+     * Calculates all exactly visible options of sections.
+     */
+    get allVisibleOptionsCount() {
```

Comment:
This is new syntax. was there a discussion on it's usage?

 We shouldn't use new code patterns without getting a buy in from most of the team because it can reduce readability.

Reply:
Got you, I will update it to methods.

Reply:
Done - 059a3025c14dce4349d744e5d3ac24c221c1b721.